src/error.rs

```diff
@@ -114,7 +116,19 @@ pub struct Error {
     /// A human-readable error message, usually a sentence explaining what went wrong.
     pub message: String,
     /// The http status code
-    pub status_code: http::StatusCode,
+    pub status_code: StatusCode,
+}
+
+impl EndpointError for Error {
+    fn try_from_response(
+        response: ruma_api::exports::http::Response<Vec<u8>>,
```

Comment:
`ruma_api::exports` is only meant to be used from ruma-api-macros.

```suggestion
        response: http::Response<Vec<u8>>,
```